### PR TITLE
MON-2900: add nodeExporter.collectors.netclass settings.

### PR DESCRIPTION
**Pull request by raptorsun**
* [x] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

This PR adds toggle for netclass collector, based on [PR#1888](https://github.com/openshift/cluster-monitoring-operator/pull/1888). It should be merged after [PR#1888](https://github.com/openshift/cluster-monitoring-operator/pull/1888).

### CODE REVIEW

**Comment by raptorsun** [comment]
/hold 
until  https://github.com/openshift/cluster-monitoring-operator/pull/1888 is merged.

File: test/e2e/node_exporter_test.go [hunk 1]
```diff
@@ -126,6 +126,14 @@ nodeExporter:
 				"node_network_transmit_packets_total",
 			},
 		},
+		{
+			nameCollector: "netclass",
```
Comment (simonpasquier) [inline test/e2e/node_exporter_test.go:130]: we need another test asserting that the netclass metrics exist in the default config.

**Comment by raptorsun** [comment]
/unhold
since https://github.com/openshift/cluster-monitoring-operator/pull/1888 is merged.

File: test/e2e/node_exporter_test.go (hunk 1 above)
Comment (raptorsun) [reply test/e2e/node_exporter_test.go:130]: metrics check is added now.

**Review by simonpasquier — COMMENTED** [review]
/lgtm

**Comment by raptorsun** [comment]
🙋‍♂️Hello @bburt-rh @Senthamilarasu-STA , could you please review this PR? If everything is good, could you stamp it with label `docs-approved`, `px-approved`?

File: Documentation/api.md [hunk 2]
```diff
@@ -192,6 +193,7 @@ The `NodeExporterCollectorConfig` resource defines settings for individual colle
 | cpufreq | [NodeExporterCollectorCpufreqConfig](#nodeexportercollectorcpufreqconfig) | Defines the configuration of the `cpufreq` collector, which collects CPU frequency statistics. Disabled by default. |
 | tcpstat | [NodeExporterCollectorTcpStatConfig](#nodeexportercollectortcpstatconfig) | Defines the configuration of the `tcpstat` collector, which collects TCP connection statistics. Disabled by default. |
 | netdev | [NodeExporterCollectorNetDevConfig](#nodeexportercollectornetdevconfig) | Defines the configuration of the `netdev` collector, which collects network devices statistics. Enabled by default. |
+| netclass | [NodeExporterCollectorNetClassConfig](#nodeexportercollectornetclassconfig) | Defines the configuration of the `netclass` collector, which collects network devices information. Enabled by default. |
```
Comment (bburt-rh) [inline Documentation/api.md:196]:
```suggestion
| netclass | [NodeExporterCollectorNetClassConfig](#nodeexportercollectornetclassconfig) | Defines the configuration of the `netclass` collector, which collects information about network devices. Enabled by default. |
```

File: Documentation/openshiftdocs/modules/nodeexportercollectorconfig.adoc [hunk 3]
```diff
@@ -24,6 +24,8 @@ Appears in: link:nodeexporterconfig.adoc[NodeExporterConfig]
 
 |netdev|link:nodeexportercollectornetdevconfig.adoc[NodeExporterCollectorNetDevConfig]|Defines the configuration of the `netdev` collector, which collects network devices statistics. Enabled by default.
 
+|netclass|link:nodeexportercollectornetclassconfig.adoc[NodeExporterCollectorNetClassConfig]|Defines the configuration of the `netclass` collector, which collects network devices information. Enabled by default.
```
Comment (bburt-rh) [inline Documentation/openshiftdocs/modules/nodeexportercollectorconfig.adoc:27]:
```suggestion
|netclass|link:nodeexportercollectornetclassconfig.adoc[NodeExporterCollectorNetClassConfig]|Defines the configuration of the `netclass` collector, which collects information about network devices. Enabled by default.
```

File: pkg/manifests/types.go [hunk 4]
```diff
@@ -267,6 +267,9 @@ type NodeExporterCollectorConfig struct {
 	// Defines the configuration of the `netdev` collector, which collects network devices statistics.
 	// Enabled by default.
 	NetDev NodeExporterCollectorNetDevConfig `json:"netdev,omitempty"`
+	// Defines the configuration of the `netclass` collector, which collects network devices information.
```
Comment (bburt-rh) [inline pkg/manifests/types.go:270]:
```suggestion
	// Defines the configuration of the `netclass` collector, which collects information about network devices.
```

**Comment by bburt-rh** [comment]
@raptorsun I added a few suggested changes.

**Comment by raptorsun** [comment]
Thank for the review @bburt-rh  :) 
The requested modifications have been pushed into the new commit.

**Review by simonpasquier — COMMENTED** [review]
/lgtm

**Comment by raptorsun** [comment]
🙋‍♂️Hello @Tai-RedHat  @Senthamilarasu-STA , could you please review this PR? If everything is good, could you stamp it with label `qe-approved`, `px-approved`?

**Comment by raptorsun** [comment]
Merge conflict is resolved. Need stamp `lgtm` again 🙋‍♂️

**Comment by Tai-RedHat** [comment]
tested PR with cluster-bot, new feature implemented.
/label qe-approved

**Comment by Tai-RedHat** [comment]
@raptorsun , shouldn't we make  `netclass` enabled by default?

```
| netclass | [NodeExporterCollectorNetClassConfig](#nodeexportercollectornetclassconfig) | Defines the configuration of the `netclass` collector, which collects information about network devices. Enabled by default. |
```

**Comment by raptorsun** [comment]
> @raptorsun , shouldn't we make `netclass` enabled by default?
> 
> ```
> | netclass | [NodeExporterCollectorNetClassConfig](#nodeexportercollectornetclassconfig) | Defines the configuration of the `netclass` collector, which collects information about network devices. Enabled by default. |
> ```

Yes, the `netclass` collector is enabled by default, retaining the same behavior as before.

**Comment by Tai-RedHat** [comment]
> > @raptorsun , shouldn't we make `netclass` enabled by default?
> > ```
> > | netclass | [NodeExporterCollectorNetClassConfig](#nodeexportercollectornetclassconfig) | Defines the configuration of the `netclass` collector, which collects information about network devices. Enabled by default. |
> > ```
> 
> Yes, the `netclass` collector is enabled by default, retaining the same behavior as before.

ok, but what you wrote in Jira Description is the default value is false.
 https://issues.redhat.com/browse/MON-2900

**Comment by raptorsun** [comment]
> > > @raptorsun , shouldn't we make `netclass` enabled by default?
> > > ```
> > > | netclass | [NodeExporterCollectorNetClassConfig](#nodeexportercollectornetclassconfig) | Defines the configuration of the `netclass` collector, which collects information about network devices. Enabled by default. |
> > > ```
> > 
> > 
> > Yes, the `netclass` collector is enabled by default, retaining the same behavior as before.
> 
> ok, but what you wrote in Jira Description is `the default value is false`. https://issues.redhat.com/browse/MON-2900

oops, my bad, it is corrected now in the JIRA ticket. 
Thank you very much :)

**Comment by raptorsun** [comment]
🙋‍♂️ Hello @Senthamilarasu-STA, the PR just needs a `px-approved` label before merging. Could you please have a look?

**Pull request merged**